src/main/java/net/sf/mzmine/parameters/impl/SimpleParameterSet.java

```diff
@@ -59,6 +60,14 @@ public SimpleParameterSet(Parameter<?> parameters[]) {
     return parameters;
   }
 
+  public void setSkipSensitiveParameters(boolean skipSensitiveParameters) {
```

Comment:
Is it useful to have the skipSensitiveParameters as a field of this class (i.e., a state that is remembered)? I would simply add it as a parameter of saveValuesToXML()

Reply:
I agree that Parameter and ParameterSet objects should be kept lightweight.

But if we just add it as a parameter of saveValuesToXML(), it is not so easy to inherit this information to Parameters that contain ParameterSets (e.g. ParameterSetParameter).

To do so we would have to add the skipSensitive boolean to the saveValueToXML() method of the Parameters.

The disadvantage of this solution is that we would have to check the skipSensitive value twice. One time in the ParameterSet (to not write an empty xml block of skipped parameters) and another time within the saveValueToXML() method of sensitive parameters. Which I found less consistent.

In general, I am fine with both solutions. Let me know which you prefer.